databuilder/backends/tpp.py

```diff
@@ -249,21 +249,42 @@ class TPPBackend(BaseBackend):
                 ethnic___9,
                 ethnic___10,
                 covid19_vaccine,
-                CAST(covid19_vaccined AS DATE) as covid19_vaccined,
-                CAST(covid19_vaccine2d AS DATE) as covid19_vaccine2d,
+                CASE
+                    WHEN covid19_vaccined = 'NA' THEN null
+                    ELSE CAST(covid19_vaccined AS DATE)
+                END AS covid19_vaccined,
+                CASE
+                    WHEN covid19_vaccine2d = 'NA' THEN null
+                    ELSE CAST(covid19_vaccine2d AS DATE)
+                END AS covid19_vaccine2d,
```

Comment:
You could write this logic slightly more neatly as:
```suggestion
                    WHEN covid19_vaccine2d != 'NA' THEN CONVERT(DATE, covid19_vaccined, 23)
                END AS covid19_vaccine2d,
```

i.e. you don't need the explicit NULL clause.